_posts/2022-04-28-upgrading-foremankatello-from-el7-to-el8.md

```diff
@@ -0,0 +1,140 @@
+   ```
+2. Install required packages:
+   ```console
+   # yum install leapp leapp-repository leapp-data-centos
+   ```
+3. Add Foreman/Katello specific repositories to `/etc/leapp/files/leapp_upgrade_repositories.repo`.
+   ```ini
+   [leapp-foreman]
+   name=Foreman 3.2
+   baseurl=https://yum.theforeman.org/releases/3.2/el8/$basearch
+   enabled=1
+   gpgcheck=1
+   gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+   module_hotfixes=1
+
+   [leapp-foreman-plugins]
+   name=Foreman plugins 3.2
+   baseurl=https://yum.theforeman.org/plugins/3.2/el8/$basearch
+   enabled=1
+   gpgcheck=0
+   gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+   module_hotfixes=1
+
+   [leapp-foreman-client]
+   name=Foreman client 3.2
+   baseurl=https://yum.theforeman.org/client/3.2/el8/$basearch
+   enabled=1
+   gpgcheck=1
+   gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman-client
+
+   [leapp-puppet7]
+   name=Puppet 7 Repository el 8 - $basearch
+   baseurl=http://yum.puppetlabs.com/puppet7/el/8/$basearch
+   gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-puppet7-release
+          file:///etc/pki/rpm-gpg/RPM-GPG-KEY-2025-04-06-puppet7-release
+   enabled=1
+   gpgcheck=1
+
+   [leapp-katello]
+   name=Katello 4.4
+   baseurl=https://yum.theforeman.org/katello/4.4/katello/el8/$basearch/
+   gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+   enabled=1
+   gpgcheck=1
+   module_hotfixes=1
+
+   [leapp-katello-candlepin]
+   name=Candlepin: an open source entitlement management system.
+   baseurl=https://yum.theforeman.org/katello/4.4/candlepin/el8/$basearch/
+   gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
+   enabled=1
+   gpgcheck=1
+   module_hotfixes=1
+
+   [leapp-pulpcore]
+   name=pulpcore: Fetch, Upload, Organize, and Distribute Software Packages.
+   baseurl=https://yum.theforeman.org/pulpcore/3.16/el8/$basearch/
+   gpgkey=https://yum.theforeman.org/pulpcore/3.16/GPG-RPM-KEY-pulpcore
+   enabled=1
+   gpgcheck=1
+   module_hotfixes=1
+   ```
+
+   * If you're not using Katello, you can leave out `leapp-katello`, `leapp-katello-candlepin` and `leapp-pulpcore`.
+   * If you don't have the `foreman-client` repository enabled today, you can leave out `leapp-foreman-client`.
+   * If you're using Puppet 6 instead of Puppet 7, adjust the `leapp-puppet7` entry accordingly (replacing `7` with `6` should work).
+   * You need a Puppet repository, even if you don't use the Puppet Server functionality, for the Puppet Agent that the installer is using.
+4. Remove `epel-release` as we don't support EL8 installations with EPEL8 enabled:
+   ```console
+   # yum remove epel-release
+   ```
+5. Remove `centos-release-scl` and `centos-release-scl-rh` repositories, as we don't need them anymore on EL8:
+   ```console
+   # yum remove centos-release-scl centos-release-scl-rh
+   ```
+6. Let LEAPP analyze your system:
+   ```console
+   # leapp preupgrade
```

Comment:
In the even this finds harder to fix issues, should it happen before users go and remove a bunch of repositories off their systems?

Reply:
what kind of issues do you envision (that would be solved by having repos)?

Reply:
I have no idea what issues all this step finds, but if its something big enough that I want to back off and keep my system as is while I investigate the changes it requires then it would be nice to have the repos? I'd be happiest if you said there is nothing that big it finds, I just know from a few times trying upgrades and scratching my head at the output.